packages/headless/src/app/commerce-ssr-engine/types/common.ts

```diff
@@ -225,3 +286,15 @@ export type SubControllerDefinitionWithProps<
       : TDefinition extends {listing: false; search: false}
         ? InvalidControllerDefinition
         : never;
+
+export type DefinedSolutionTypes<
```

Comment:
The idea of adding this type is to infer which `SolutionType` values are valid when calling the `build` function, but upon further reflexion since this function is only used internally I think we could get rid of it after all. Commerce SSR typing is already complicated enough as it is 😅

Reply:
I'm pushing a commit that reverts the type-related changes.